tt_metal/impl/device/device.cpp

```diff
@@ -559,13 +559,15 @@ void Device::update_workers_build_settings(std::vector<std::vector<std::tuple<tt
                 uint32_t num_dispatchers = device_worker_variants[DISPATCH].size();
                 TT_ASSERT(device_worker_variants[DEMUX].size() == 1, "Cannot have more than one Demux.");
                 auto demux_settings = std::get<1>(device_worker_variants[DEMUX][0]);
+                auto prefetch_h_settings = std::get<1>(device_worker_variants[PREFETCH][0]);
+                auto prefetch_physical_core = prefetch_h_settings.worker_physical_core
                 TT_ASSERT(num_dispatchers == demux_settings.semaphores.size(), "Demux does not have required number of semaphores for Dispatchers. Exptected = {}. Fount = {}", num_dispatchers, demux_settings.semaphores.size());
                 uint32_t demux_sem = demux_settings.producer_semaphore_id;
                 for (auto&[core, settings] : device_worker_variants[DISPATCH]) {
                     auto dispatch_core_type = settings.dispatch_core_type;
                     settings.upstream_cores.push_back(demux_settings.worker_physical_core);
                     settings.downstream_cores.push_back(tt_cxy_pair(0, 0, 0));
-                    settings.compile_args.resize(17);
+                    settings.compile_args.resize(21);
```

Comment:
what is this magic number?

Reply:
This is the size of the preallocated compile args vector sent to to dispatch core. We had to add 4 more compile args, hence 17 -> 24.

Reply:
Can we de magicify this?